apps/dgst.c

```diff
@@ -497,9 +507,12 @@ int do_fp(BIO *out, unsigned char *buf, BIO *bp, int sep, int binout,
     if (binout) {
         BIO_write(out, buf, len);
     } else if (sep == 2) {
-        file = newline_escape_filename(file);
+        file = newline_escape_filename(file, &backslash);
 
         for (i = 0; i < (int)len; i++)
+            if (i == 0 && backslash == 1)
+                BIO_printf(out, "\\%02x", buf[i]);
+            else
```

Comment:
Wouldn't it be easier if you have this before the loop?

``` C
        if (backslash != 0)
            BIO_puts(out, "\\");
```

Reply:
Great suggestion.  This will be cleaner than checking on each iteration of the for loop.  Thanks.  Updated on my latest commit.